File: CNS Lab (Sem 7)/Python Equivalent/hill.py
import numpy as np

K=np.array([[17,17,5],[21,18,21],[2,2,19]])
IK=np.array([[4,9,15],[15,17,6],[24,0,17]])

m=input("Enter a 3-letter lowercase word: ")
v=np.array([ord(c)-97 for c in m])

e=K.dot(v)%26
print("Encrypted:",''.join(chr(i+97) for i in e))

d=IK.dot(e)%26
print("Decrypted:",''.join(chr(i+97) for i in d))

File: Sem 7 CNS Lab/hill.m
function hill_cipher()
    clc;
    fprintf('--- SIMPLE HILL CIPHER (3x3) ---\n\n');

    key = [17, 17, 5;
           21, 18, 21;
           2, 2, 19];

    inv_key = [4, 9, 15;
               15, 17, 6;
               24, 0, 17];

    msg = input('Enter a 3-letter lowercase word (e.g., "cat"): ', 's');

    if length(msg) ~= 3
        error('Error: Please enter exactly 3 letters.');
    end

    msg_numbers = double(msg) - 97;

    msg_vector = msg_numbers';

    fprintf('\nOriginal numbers: %d %d %d\n', msg_numbers);

    encrypt_math = key * msg_vector;
    encrypt_vec  = mod(encrypt_math, 26);

    encrypted_msg = char(encrypt_vec' + 97);

    fprintf('Encrypted word:   %s\n', encrypted_msg);

    decrypt_math = inv_key * encrypt_vec;
    decrypt_vec  = mod(decrypt_math, 26);

    decrypted_msg = char(decrypt_vec' + 97);

    fprintf('Decrypted word:   %s\n', decrypted_msg);
end
